common/test/acceptance/tests/lms/test_learner_profile.py

```diff
@@ -0,0 +1,259 @@
+
+    def test_fields_on_others_public_profile(self):
+        """
+        Scenario: Verify that desired fields are shown when looking at her own public profile.
+
+        Given that I am a registered user.
+        And I visit My Profile page.
+        And I set the profile visibility to public.
+        And I reload the page.
+        Then I shouldn't see the profile visibility selector dropdown.
+        Then I see all the profile fields are shown.
+        Also `location`, `language` and `about me` fields are not editable.
+        """
+        self.visit_other_profile_page(self.OTHER_USER, privacy=self.PRIVACY_PUBLIC)
+        self.visit_other_profile_page(self.MY_USER)
+
+        self.assertFalse(self.other_profile_page.privacy_field_visible)
+
+        # We are excluding language field from verification because when a usr view another users profile,
+        # server send `languages` field in model instead of `language`, due to which language field will not be shown
+        # Until this is fixed on server side, we will exclude the language fields.
+        fields_to_check = self.PUBLIC_PROFILE_FIELDS[0:2] + self.PUBLIC_PROFILE_FIELDS[3:]
+        self.assertEqual(self.other_profile_page.visible_fields, fields_to_check)
+
+        self.assertEqual(self.my_profile_page.editable_fields, [])
+
+    def _test_dropdown_field(self, field_id, new_value, displayed_value, mode):
+        """
+        Test behaviour of a dropdown field.
+        """
+        self.visit_my_profile_page(self.MY_USER, privacy=self.PRIVACY_PUBLIC)
+
+        self.my_profile_page.value_for_dropdown_field(field_id, new_value)
+        self.assertEqual(self.my_profile_page.get_non_editable_mode_value(field_id), displayed_value)
+        self.assertTrue(self.my_profile_page.mode_for_field(field_id), mode)
+
+        self.browser.refresh()
+        self.my_profile_page.wait_for_page()
+
+        self.assertEqual(self.my_profile_page.get_non_editable_mode_value(field_id), displayed_value)
+        self.assertTrue(self.my_profile_page.mode_for_field(field_id), mode)
+
+    def _test_textarea_field(self, field_id, new_value, displayed_value, mode):
+        """
+        Test behaviour of a textarea field.
+        """
+        self.visit_my_profile_page(self.MY_USER, privacy=self.PRIVACY_PUBLIC)
+
+        self.my_profile_page.value_for_textarea_field(field_id, new_value)
+        self.assertEqual(self.my_profile_page.get_non_editable_mode_value(field_id), displayed_value)
+        self.assertTrue(self.my_profile_page.mode_for_field(field_id), mode)
+
+        self.browser.refresh()
+        self.my_profile_page.wait_for_page()
+
+        self.assertEqual(self.my_profile_page.get_non_editable_mode_value(field_id), displayed_value)
+        self.assertTrue(self.my_profile_page.mode_for_field(field_id), mode)
+
+    def test_country_field(self):
+        """
+        Test behaviour of `Country` field.
```

Comment:
Please make these comments into BDD-style specs like the ones above. I thought these were helper methods initially. I'd also reload the page in each instance and verify that the value has persisted.